lib/llm/src/local_model.rs

```diff
@@ -0,0 +1,164 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use anyhow::Context;
+use dynamo_runtime::component::Endpoint;
+use dynamo_runtime::traits::DistributedRuntimeProvider;
+
+use crate::http::service::discovery::{ModelEntry, ModelNetworkName};
+use crate::key_value_store::{EtcdStorage, KeyValueStore, KeyValueStoreManager};
+use crate::model_card::{self, ModelDeploymentCard};
+use crate::model_type::ModelType;
+
+/// Prefix for Hugging Face model repository
+const HF_SCHEME: &str = "hf://";
+
+/// What we call a model if the user didn't provide a name. Usually this means the name
+/// is invisible, for example in a text chat.
+const DEFAULT_NAME: &str = "dynamo";
```

Comment:
Just curious - would anything unexpected happen if a user set a model name to "dynamo" (conflicting with DEFAULT_NAME)? Wasn't sure if there were any assumptions about this being a reserved/internal name elsewhere.

Reply:
It would be fine. This is for the text chat, where the user isn't sending json with `"name": "model-name"`, so they don't need to know the model name. If the backend also doesn't require a name (the echo engines), then it doesn't matter so we use this internally.